packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.ts

```diff
@@ -108,11 +121,13 @@ const DEFAULT_EXPANDED_ITEMS: string[] = [];
 useTreeViewExpansion.getDefaultizedParams = (params) => ({
   ...params,
   defaultExpandedItems: params.defaultExpandedItems ?? DEFAULT_EXPANDED_ITEMS,
+  expansionTrigger: params.expansionTrigger ?? 'content',
```

Comment:
I think you need to remove this default value.
Otherwise you won't be able to know if the `params.expansionTrigger` comes from the user or is the default value when deciding what to do with editing.